### PR TITLE
refactor: consolidate patient id parameter handling across per-endpoint handlers

### PR DESCRIPTION
**Pull request by bradymiller**
## Summary

Consolidates patient id parameter handling across 11 per-endpoint handlers that previously read `$_GET['pid']` / `$_POST['pid']` as untyped mixed values, or relied on ambient session bootstrap for pid context. Each endpoint now takes explicit responsibility for its pid parameter: narrow the input to `int` at entry, gate the handler by the role required for the action, and use the resulting `$pid` consistently for downstream queries.

Extends the pattern already established by `interface/patient_file/encounter/load_form.php` (form ownership verification via `EncounterFormAccess::isFormOwnedBySession`) and by `interface/patient_file/summary/demographics.php` (explicit `setpid()` bootstrap for URL-linked patient context).

## Files touched

- `interface/billing/ub04_dispose.php` — role gate + int-narrow pid/encounter on each handler branch
- `interface/billing/ub04_form.php` — role gate + deep-linked `setpid()` bootstrap
- `interface/forms/CAMOS/notegen.php` — role gate at top
- `interface/forms/eye_mag/SpectacleRx.php` — role gate + collapse `$_REQUEST` fallback chain + add missing form-token check on 3 state-changing branches + emit form token in AJAX form + remove-dispense payload
- `interface/forms/eye_mag/save.php` — role gate + fix latent bug where `$id` fell back to `$_REQUEST['pid']` (form id and pid are distinct values); scope tracker lookup to session pid
- `interface/forms/vitals/growthchart/chart.php` — role gate + deep-linked bootstrap + drop redundant `isset+is_numeric` guard
- `interface/modules/custom_modules/oe-module-weno/templates/indexrx.php` — require patient context; normalize scattered `$session->get('pid') ?? $pid` reads to a single `$pid` var
- `interface/patient_file/birthday_alert/birthday_pop.php` — role gate + deep-linked bootstrap + string cast for `js_escape()`
- `interface/patient_file/encounter/view_form.php` — mirror `load_form.php`: fetch form owner, verify ownership + sensitivity gate before dispatching
- `interface/procedure_tools/libs/labs_ajax.php` — role gate + int-narrow pid with early-exit on invalid
- `library/ajax/easipro_util.php` — hoist `$pid` assignment above dispatch chain; role gate on `request_assessment` and `record_result`; pre-flight ownership check on `record_result`

Baseline reductions (8 files under `.phpstan/baseline/`): entries removed where the underlying pattern was fixed by this refactor (e.g. `$_REQUEST` count reductions, `empty()` replaced with strict comparison, `cast.int` sites narrowed). No new baseline entries added.

## Notable behavior change

- `interface/forms/eye_mag/save.php` patient_tracker block: the swap from `sqlFetchArray(sqlStatement(...))` to `QueryUtils::querySingleRow(...)` + `if ($tracker !== false)` guard fixes a latent orphan-INSERT bug. Previously when no tracker row matched the `pid`+`apptdate` SELECT, the follow-on INSERT into `patient_tracker_element` still ran with `pt_tracker_id = NULL` from `$tracker['id']` (the two UPDATEs on either side affected 0 rows via `WHERE id = null` / `WHERE pc_eid = null`, so the orphan insert was the only visible side-effect). The new guard skips the whole block cleanly when there is no matching tracker row — the intended semantics for a "record status change on today's appointment" flow when no such appointment exists. Behavior for the matching-tracker path is unchanged.

## Test plan

- [ ] UB04 claim: save/reset flows with a valid patient context
- [ ] eye_mag Rx: print + save + dispense update
- [ ] Growth chart: deep-link from vitals form
- [ ] Encounter form view: opening a `sensitivities.high` encounter without `sensitivities.high` permission renders the standard denial page
- [ ] Birthday alert popup + turnoff
- [ ] CAMOS note generator; labs label print
- [ ] Weno eRx compose flow (if Weno enabled)
- [ ] Easipro PROM request + record result (if Easipro enabled)

### CODE REVIEW

File: interface/modules/custom_modules/oe-module-weno/templates/indexrx.php [hunk 1]
```diff
@@ -186,7 +194,7 @@ function generateDebugInfo() {
             <form>
                 <header class="bg-light text-dark text-center">
                     <h3>
-                        <a href="<?php echo OEGlobalsBag::getInstance()->getWebRoot() ?>/interface/patient_file/summary/demographics.php?resync=true&set_pid=<?php echo urlencode(attr($session->get('pid') ?? $pid)) ?>" class="text-primary" title="<?php echo xla("Return to Patient Demographics"); ?>"><?php echo xlt("e-Prescribe"); ?>
+                        <a href="<?php echo OEGlobalsBag::getInstance()->getWebRoot() ?>/interface/patient_file/summary/demographics.php?resync=true&set_pid=<?php echo urlencode(attr((string) $pid)) ?>" class="text-primary" title="<?php echo xla("Return to Patient Demographics"); ?>"><?php echo xlt("e-Prescribe"); ?>
```
Comment (kojiromike) [inline interface/modules/custom_modules/oe-module-weno/templates/indexrx.php]: It isn't necessary to sanitize or urlencode an integer.

File: interface/patient_file/birthday_alert/birthday_pop.php [hunk 2]
```diff
@@ -15,12 +15,33 @@
...
+// handler (would change active patient without a form-token round-trip). The
+// AJAX turnoff below re-sends pid in a form-token-protected POST; the turnoff
+// handler (library/ajax/turnoff_birthday_alert.php) validates the token but
+// does not currently compare the submitted pid against the session pid --
+// tightening that comparison is a separate improvement outside this refactor.
+$query = CurrentRequest::get()->query;
+$pid = $query->getInt('pid');
+if ($pid <= 0) {
+    echo "<p>" . xlt('Missing PID.') . "</p>";
+    exit;
```
Comment (kojiromike) [inline interface/patient_file/birthday_alert/birthday_pop.php]: Please consider the [RequestTerminator](https://github.com/openemr/openemr/blob/master/src/Common/Http/RequestTerminator.php) so that the right http status code gets sent.

File: interface/procedure_tools/libs/labs_ajax.php [hunk 3]
```diff
@@ -64,18 +71,19 @@ function orderDate($order)
 }
 
 if ($action === 'print_labels') {
+    $query = CurrentRequest::get()->query;
     $client = $_GET['acctid'];
-    $pid = $_GET['pid'];
+    $pid = $query->getInt('pid');
+    if ($pid <= 0) {
+        exit;
```
Comment (kojiromike) [inline interface/procedure_tools/libs/labs_ajax.php]: https://github.com/openemr/openemr/blob/master/src/Common/Http/RequestTerminator.php please

File: interface/billing/ub04_dispose.php [hunk 4]
```diff
@@ -14,35 +14,97 @@
...
+ * Called from `ub04_form.php` (page render) and `ub04_submit.php` (AJAX
+ * endpoint). Returns early when no `handler` is present; otherwise gates on
+ * the `acct/bill` write role, validates the form token, requires the
+ * submitted pid to match the session patient (for pid-scoped handlers), and
+ * dispatches to one of:
+ *   - `edit_save`   -> saveTemplate() for the pid/encounter
+ *   - `payer_save`  -> savePayerTemplate() for a payer id (no pid context)
+ *   - `batch_save`  -> saveTemplate() in batch mode
+ *   - `reset_claim` -> BillingUtilities::updateClaim() clearing the claim
+ * Every dispatched branch terminates the request with exit(); the function
```
Comment (kojiromike) [inline interface/billing/ub04_dispose.php]: We should not use `exit()` anymore, anywhere.

File: interface/billing/ub04_dispose.php [hunk 5]
```diff
@@ -14,35 +14,97 @@
...
-            $encounter = $_POST['encounter'] ?? $_GET['encounter'];
-            $action = $_REQUEST['action'];
-            $ub04id = json_decode((string) $ub04id, true);
-            saveTemplate($encounter, $pid, $ub04id, $action);
+            $ub04id = $request->getString('ub04id');
+            $pid = $request->getInt('pid');
+            $encounter = $request->getInt('encounter');
+            $action = $request->getString('action');
+            if ($pid <= 0 || $encounter <= 0) {
+                exit();
```
Comment (kojiromike) [inline interface/billing/ub04_dispose.php]: 🙏

**Review by kojiromike — COMMENTED** [review]
I won't call out all the places this introduces a new `exit()`, but please use another pathway that returns the right http status code and is more debuggable.

**Comment by bradymiller** [comment]
@kojiromike , :pray: answered :)
